src/index.tsx

```diff
@@ -237,57 +252,65 @@ function App() {
-            onClick={onMapClickHandler}
-            onHover={onMapHoverHandler}
-            useDevicePixels={
-              isDefined(useDevicePixels) ? useDevicePixels : true
-            }
-            // https://deck.gl/docs/api-reference/core/deck#_typedarraymanagerprops
-            _typedArrayManagerProps={{
-              overAlloc: 1,
-              poolSize: 0,
+            mapStyle={mapStyle || DEFAULT_MAP_STYLE}
+            attributionControl={{
+              customAttribution,
             }}
-            onViewStateChange={(event) => {
-              const { viewState } = event;
-
-              // This condition is necessary to confirm that the viewState is
-              // of type MapViewState.
-              if ("latitude" in viewState) {
-                const { longitude, latitude, zoom, pitch, bearing } = viewState;
-                setViewState({
-                  longitude,
-                  latitude,
-                  zoom,
-                  pitch,
-                  bearing,
-                });
-              }
-            }}
-            parameters={parameters || {}}
           >
-            <Map
-              mapStyle={mapStyle || DEFAULT_MAP_STYLE}
-              customAttribution={customAttribution}
-            ></Map>
-          </DeckGL>
+            <FullscreenControl position="top-left" />
+            <NavigationControl position="top-left" />
+            <ScaleControl position="bottom-left" />
+            <DeckGLOverlay
+              // interleaved
+              style={{ width: "100%", height: "100%" }}
+              layers={
+                bboxSelectPolygonLayer
+                  ? layers.concat(bboxSelectPolygonLayer)
+                  : layers
+              }
+              getTooltip={(showTooltip && getTooltip) || undefined}
+              getCursor={() => (isDrawingBBoxSelection ? "crosshair" : "grab")}
+              pickingRadius={pickingRadius}
+              onClick={onMapClickHandler}
+              onHover={onMapHoverHandler}
+              useDevicePixels={
+                isDefined(useDevicePixels) ? useDevicePixels : true
+              }
+              // https://deck.gl/docs/api-reference/core/deck#_typedarraymanagerprops
+              _typedArrayManagerProps={{
+                overAlloc: 1,
+                poolSize: 0,
+              }}
+              onViewStateChange={(event) => {
+                const { viewState } = event;
+
+                // This condition is necessary to confirm that the viewState is
+                // of type MapViewState.
+                if ("latitude" in viewState) {
+                  const { longitude, latitude, zoom, pitch, bearing } =
+                    viewState;
+                  setViewState({
+                    longitude,
+                    latitude,
+                    zoom,
+                    pitch,
+                    bearing,
+                  });
+                }
+              }}
```

Comment:
todo: I need to check that deck.gl still sends these events even when the view state is managed by maplibre. 

For background: This sets the current view state on the _model_ so that it can be updated on the Python side, which can be accessed via `Map.view_state`.